cli/azd/pkg/account/subscriptions_manager.go

```diff
@@ -303,23 +305,23 @@ func (m *SubscriptionsManager) GetSubscription(ctx context.Context, subscription
 	return &sub, nil
 }
 
-func toSubscriptions(azSubs []azcli.AzCliSubscriptionInfo, userAccessTenantId string) []Subscription {
+func toSubscriptions(azSubs []*armsubscriptions.Subscription, userAccessTenantId string) []Subscription {
 	if azSubs == nil {
 		return nil
 	}
 
 	subs := make([]Subscription, 0, len(azSubs))
 	for _, azSub := range azSubs {
-		subs = append(subs, toSubscription(azSub, userAccessTenantId))
+		subs = append(subs, toSubscription(*azSub, userAccessTenantId))
 	}
 	return subs
 }
 
-func toSubscription(subscription azcli.AzCliSubscriptionInfo, userAccessTenantId string) Subscription {
+func toSubscription(subscription armsubscriptions.Subscription, userAccessTenantId string) Subscription {
 	return Subscription{
-		Id:                 subscription.Id,
-		Name:               subscription.Name,
-		TenantId:           subscription.TenantId,
+		Id:                 *subscription.SubscriptionID,
+		Name:               *subscription.DisplayName,
+		TenantId:           *subscription.TenantID,
```

Comment:
Do we know if this is safe?

Reply:
I'm a little worried about the unsafe dereference everywhere.

This is maybe a hint of what I meant in #1577. If we don't safely discard the request at the client level, and instead propagate the client contract everywhere, then we need nil-checks everywhere. Perhaps this invariant is satisfied at runtime, but I'm still worried that one of these things are unsafe -- there's no way to know for sure without type safetyness.

I think we can go in two different ways:
1. Option 1: `nil` check everywhere
1. Option 2: convert contract that is safe after validating "`azd` required" fields, discard invalid data

Reply:
I went down the option 2 path.  I updated the commit to expand `account.Location` so that it contains the fields from `armsubscriptions.Location` we care about and then handle the conversion of the model types in the account package.

In practice I believe that all the dereferences were safe, but for the ones where I could give back some reasonable fallback, I added the code to do that.

Reply:
Thanks for addressing it!